src/plugins/media/nodeviews/mediaNodeView/media.tsx

```diff
@@ -156,6 +163,21 @@ export class MediaNode extends Component<MediaNodeProps, MediaNodeState> {
     }
   }
 
+  fetchCozyImage = async () => {
+    try {
+      const server = await serviceClient.getImage(this.props.node.attrs.url)
+
+      if (!server?.data) throw Error('Could not fetch image from back-end')
+
+      this.setState(
+        { imageUrl: `${stackClient.uri}${server.data.links.small}` },
+        this.forceUpdate
+      )
+    } catch (error) {
+      this.setState({ imageError: error.message }, this.forceUpdate)
```

Comment:
why it's necessary the `forceUpdate` ?

Reply:
Yeah it is because of the SCU method. I figured I didn't want to modify it as it's already huge, and just forceUpdate my way through, less LoC. What you think?
 ```
shouldComponentUpdate(nextProps: MediaNodeProps, nextState: MediaNodeState) {
    const hasNewViewMediaClientConfig =
      !this.state.viewMediaClientConfig && nextState.viewMediaClientConfig
    if (
      this.props.selected !== nextProps.selected ||
      this.props.uploadComplete !== nextProps.uploadComplete ||
      this.props.node.attrs.id !== nextProps.node.attrs.id ||
      this.props.node.attrs.collection !== nextProps.node.attrs.collection ||
      this.props.maxDimensions.height !== nextProps.maxDimensions.height ||
      this.props.maxDimensions.width !== nextProps.maxDimensions.width ||
      this.props.contextIdentifierProvider !==
        nextProps.contextIdentifierProvider ||
      this.props.isLoading !== nextProps.isLoading ||
      this.props.mediaProvider !== nextProps.mediaProvider ||
      hasNewViewMediaClientConfig
    ) {

      return true
    }
    return false
  }
  ```

Reply:
Je me dis qu'on est plus à une ligne prêt :D et au moins ça sera au même endroit (AMHA plus facile à debug si on cherche pk ça re-render ou pas).

Reply:
updated